packages/muelu/src/Transfers/SemiCoarsen/MueLu_SemiCoarsenPFactory_decl.hpp

```diff
@@ -145,7 +145,7 @@ namespace MueLu {
     LO MakeSemiCoarsenP(LO const Ntotal, LO const nz, LO const CoarsenRate, LO const LayerId[],
                                   LO const VertLineId[], LO const DofsPerNode, RCP<Matrix>& Amat,
                                   RCP<Matrix>& P, RCP<const Map>& coarseMap, 
-                                  const RCP<MultiVector> fineNullspace, RCP<MultiVector>& coarseNullspace, RCP<Matrix>& R, bool buildRestriction) const;
+                                  const RCP<MultiVector> fineNullspace, RCP<MultiVector>& coarseNullspace, RCP<Matrix>& R, bool buildRestriction, bool doLinear) const;
```

Comment:
this might be a silly nitpick, should we have something more like `int piecewiseInterpolationDegree` instead? I only mention that in case there's some sick world where we decide to do quadratic, cubic, etc., so we won't have to change this interface again.

Reply:
This might be a possibility, but I guess we would have to use something like -1 to indicate that this type of geometric interpolation is not being used

Reply:
That's fair. It's not a huge deal either way. I've found myself on the receiving end of backwards-incompatible changes in Trilinos quite a bit, so that's the only reason I bring it up, although I'm pretty optimistic that nobody would use MueLu internal functions like this one.